stackslib/src/net/api/get_tenures_fork_info.rs

```diff
@@ -0,0 +1,361 @@
+    #[serde(with = "prefix_hex")]
+    pub sortition_id: SortitionId,
+    /// The parent of this burn block's Sortition ID
+    #[serde(with = "prefix_hex")]
+    pub parent_sortition_id: SortitionId,
+    /// The consensus hash of the block that triggered this event. This incorporates
+    ///  PoX forking information and burn op information to obtain an identifier that is
+    ///  unique across PoX forks and burnchain forks.
+    #[serde(with = "prefix_hex")]
+    pub consensus_hash: ConsensusHash,
+    /// Boolean indicating whether or not there was a succesful sortition (i.e. a winning
+    ///  block or miner was chosen).
+    pub was_sortition: bool,
+    /// If the sortition occurred, and a block was mined during the tenure, this is the
+    /// tenure's block.
+    #[serde(with = "prefix_opt_hex")]
+    pub first_block_mined: Option<StacksBlockId>,
+}
+
+mod prefix_opt_hex {
+    pub fn serialize<S: serde::Serializer, T: std::fmt::Display>(
+        val: &Option<T>,
+        s: S,
+    ) -> Result<S::Ok, S::Error> {
+        match val {
+            Some(ref some_val) => {
+                let val_str = format!("0x{some_val}");
+                s.serialize_some(&val_str)
+            }
+            None => s.serialize_none(),
+        }
+    }
+
+    pub fn deserialize<'de, D: serde::Deserializer<'de>, T: super::HexDeser>(
+        d: D,
+    ) -> Result<Option<T>, D::Error> {
+        let opt_inst_str: Option<String> = serde::Deserialize::deserialize(d)?;
+        let Some(inst_str) = opt_inst_str else {
+            return Ok(None);
+        };
+        let Some(hex_str) = inst_str.get(2..) else {
+            return Err(serde::de::Error::invalid_length(
+                inst_str.len(),
+                &"at least length 2 string",
+            ));
+        };
+        let val = T::try_from(&hex_str).map_err(serde::de::Error::custom)?;
+        Ok(Some(val))
+    }
+}
+
+mod prefix_hex {
+    pub fn serialize<S: serde::Serializer, T: std::fmt::Display>(
+        val: &T,
+        s: S,
+    ) -> Result<S::Ok, S::Error> {
+        s.serialize_str(&format!("0x{val}"))
+    }
+
+    pub fn deserialize<'de, D: serde::Deserializer<'de>, T: super::HexDeser>(
+        d: D,
+    ) -> Result<T, D::Error> {
+        let inst_str: String = serde::Deserialize::deserialize(d)?;
+        let Some(hex_str) = inst_str.get(2..) else {
+            return Err(serde::de::Error::invalid_length(
+                inst_str.len(),
+                &"at least length 2 string",
+            ));
+        };
+        T::try_from(&hex_str).map_err(serde::de::Error::custom)
+    }
+}
+
+trait HexDeser: Sized {
+    fn try_from(hex: &str) -> Result<Self, HexError>;
+}
+
+macro_rules! impl_hex_deser {
+    ($thing:ident) => {
+        impl HexDeser for $thing {
+            fn try_from(hex: &str) -> Result<Self, HexError> {
+                $thing::from_hex(hex)
+            }
+        }
+    };
+}
+
+impl_hex_deser!(BurnchainHeaderHash);
```

Comment:
Can we put the all of the infrastructure pertaining to hex serialization into `src/net/api/mod.rs`?

Reply:
Sure

Reply:
Updated in fdcaea8b91972ad721d9d9e0579ac89fb531ddc0